src/main/java/net/minecraftforge/fmp/block/BlockCoverable.java

```diff
@@ -0,0 +1,730 @@
+    {
+        return super.getPickBlock(state, target, world, pos, player);
+    }
+
+    private IMicroblockContainerTile brokenTile = null;
+    private boolean harvestingWrapper = false;
+
+    @Override
+    public final void harvestBlock(World world, EntityPlayer player, BlockPos pos, IBlockState state, TileEntity te, ItemStack stack)
+    {
+        brokenTile = te instanceof IMicroblockContainerTile ? (IMicroblockContainerTile) te : null;
+        harvestBlockDefault(world, player, pos, state, te, stack);
+        brokenTile = null;
+    }
+
+    public void harvestBlockDefault(World worldIn, EntityPlayer player, BlockPos pos, IBlockState state, TileEntity te, ItemStack stack)
+    {
+        super.harvestBlock(worldIn, player, pos, state, te, stack);
+    }
+
+    @Override
+    public final List<ItemStack> getDrops(IBlockAccess world, BlockPos pos, IBlockState state, int fortune)
+    {
+        if (harvestingWrapper) { return getDropsDefault(world, pos, state, fortune); }
+        IMicroblockContainerTile brokenTile = getMicroblockTile(world, pos);
+        if (brokenTile == null)
+        {
+            brokenTile = this.brokenTile;
+        }
+        if (brokenTile == null) { return getDropsDefault(world, pos, state, fortune); }
+        List<ItemStack> drops = new ArrayList<ItemStack>();
+        drops.addAll(getDropsDefault(world, pos, state, fortune));
+        return drops;
+    }
+
+    public List<ItemStack> getDropsDefault(IBlockAccess world, BlockPos pos, IBlockState state, int fortune)
+    {
+        return super.getDrops(world, pos, state, fortune);
+    }
+
+    @Override
+    public final boolean removedByPlayer(IBlockState state, World world, BlockPos pos, EntityPlayer player, boolean willHarvest)
+    {
+        RayTraceResult hit = reTraceAll(world, pos, player);
+        if (hit instanceof RayTraceResult)
+        {
+            IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+            if (tile != null)
+            {
+                tile.getMicroblockContainer().getPartContainer().harvest(player, (RayTraceResult) hit);
+            }
+            return false;
+        }
+        else
+        {
+            IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+            MultipartContainer container = tile != null ? tile.getMicroblockContainer().getPartContainer() : null;
+            if (container.getParts().isEmpty())
+            {
+                return removedByPlayerDefault(state, world, pos, player, willHarvest);
+            }
+            else
+            {
+                if (!removedByPlayerDefault(state, world, pos, player, willHarvest)) { return false; }
+                world.removeTileEntity(pos);
+                if (!world.setBlockState(pos, ForgeMultipart.multipart.getDefaultState(), 0)) { return false; }
+                world.removeTileEntity(pos);
+                world.setTileEntity(pos, new TileMultipartContainer(container));
```

Comment:
Removing it twice to add it again, yes... but no. The TE as it is created by default has a new multipart container. What I want to do is clone the old one onto the one that's just been created. I'll use a copyFrom() method instead, I guess. May be a bit better for memory, though here I was just reusing the instance.